UnrealPlugin/Source/PlayfabGSDK/Private/FGSDKInternal.cpp

```diff
@@ -0,0 +1,386 @@
+
+	TArray<TSharedPtr<FJsonValue>> ConnectedPlayersJson;
+	for (const FConnectedPlayer& ConnectedPlayer: HeartbeatRequest.ConnectedPlayers)
+	{
+		TSharedPtr<FJsonObject> ConnectedPlayerJsonObject = MakeShared<FJsonObject>();
+
+		ConnectedPlayerJsonObject->SetStringField(TEXT("PlayerId"), ConnectedPlayer.PlayerId);
+		
+		TSharedPtr<FJsonValueObject> ConnectedPlayerJson = MakeShared<FJsonValueObject>(ConnectedPlayerJsonObject);
+		ConnectedPlayersJson.Add(ConnectedPlayerJson);
+	}
+
+	HeartbeatRequestJson->SetArrayField(TEXT("CurrentPlayers"), ConnectedPlayersJson);
+
+	FString HeartbeatRequestJsonString;
+	TSharedRef< TJsonWriter<> > Writer = TJsonWriterFactory<>::Create(&HeartbeatRequestJsonString);
+	FJsonSerializer::Serialize(HeartbeatRequestJson.ToSharedRef(), Writer);
+
+	return HeartbeatRequestJsonString;
+}
+
+void FGSDKInternal::DecodeHeartbeatResponse(const FString& ResponseJson)
+{
+	TSharedPtr<FJsonObject> HeartbeatResponseJson;
+	TSharedRef<TJsonReader<>> Reader = TJsonReaderFactory<>::Create(ResponseJson);
+
+	if (!FJsonSerializer::Deserialize(Reader, HeartbeatResponseJson))
+	{
+		UE_LOG(LogPlayfabGSDK, Error, TEXT("Failed to parse heartbeat"));
+		return;
+	}
+
+	if (HeartbeatResponseJson->HasField(TEXT("sessionConfig")))
+	{
+		FScopeLock ScopeLock(&ConfigMutex);
+
+		TSharedPtr<FJsonObject> SessionConfigJson = HeartbeatResponseJson->GetObjectField(TEXT("sessionConfig"));
+		for (const auto& SessionConfigJsonValue: SessionConfigJson->Values)
+		{
+			FString ValueString;
+			if (SessionConfigJsonValue.Value->TryGetString(ValueString))
+			{
+				if (ConfigSettings.Contains(SessionConfigJsonValue.Key))
+				{
+					ConfigSettings[SessionConfigJsonValue.Key] = ValueString;
+				}
+				else
+				{
+					ConfigSettings.Add(SessionConfigJsonValue.Key, ValueString);
+				}
+			}
+		}		
+
+		// Update initial players only if this is the first time populating it.
+		if (InitialPlayers.Num() == 0 && SessionConfigJson->HasField(TEXT("initialPlayers")))
+		{
+			TArray<TSharedPtr<FJsonValue>> PlayersJson = SessionConfigJson->GetArrayField(TEXT("initialPlayers"));
+
+			for (auto PlayerJson: PlayersJson)
+			{
+				InitialPlayers.Add(PlayerJson->AsString());
+			}
+		}
+
+		if (SessionConfigJson->HasField(TEXT("metadata")))
+		{
+			TSharedPtr<FJsonObject> MetaDatasJson = SessionConfigJson->GetObjectField(TEXT("metadata"));
+
+			for (const auto& MetaDataJson: MetaDatasJson->Values)
+			{
+				FString ValueString;
+				if (MetaDataJson.Value->TryGetString(ValueString))
+				{
+					if (ConfigSettings.Contains(MetaDataJson.Key))
+					{
+						ConfigSettings[MetaDataJson.Key] = ValueString;
+					}
+					else
+					{
+						ConfigSettings.Add(MetaDataJson.Key, ValueString);
+					}
+				}
+			}
+		}
+	}
+
+	if (HeartbeatResponseJson->HasField("nextScheduledMaintenanceUtc"))
+	{
+		FDateTime NextMaintenance = ParseDate(HeartbeatResponseJson->GetStringField(TEXT("nextScheduledMaintenanceUtc")));
```

Comment:
What is the format of the date time string i am getting from the JSON, so I can parse it correctly?

Reply:
This string originates from here (https://docs.microsoft.com/en-us/azure/virtual-machines/windows/scheduled-events#query-for-events), should be in UTC format

Reply:
Thank you, I have adapted the code to parse it correctly